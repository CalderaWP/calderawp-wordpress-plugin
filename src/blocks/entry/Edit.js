// @flow
import React, {Fragment} from 'react';
import type {GutenbergEditProps} from "../types";
import {InspectorControls} from "@wordpress/editor";
import {EntryEdit,
	EntryDisplay,
	EntryInlineEdit,
	EntryControlsEdit
} from '../../components/Entry';
import * as cfApi from "@caldera-labs/api-client";
import type {KeyedFormCollection} from "../../flow-types/formType";

const API_URL = 'http://localhost:8218/wp-json/cf-api/v2';
const NONCE = false;
const formsAdminApiClient = new cfApi.WpClient(API_URL );
formsAdminApiClient.setCorsMode(true);
const entryApiClient =  new cfApi.EntriesClient(API_URL );
entryApiClient.setCorsMode(true);

function getForms(page:number) : Promise<any>{
	return formsAdminApiClient.makeRequest( 'forms', {full:true,page});
}

function getEntries(formId:string,page:number) : Promise<any>{
	return entryApiClient.makeRequest( `entries/${formId}`, {page});
}

function getEntry( formId:string,entryId:number): Promise<any>{
	return entryApiClient.makeRequest( `entries/${formId}/${entryId}`);
}


export default function Edit(props: GutenbergEditProps) {
	const {attributes, setAttributes, isSelected, className,id} = props;
	const {entryId,formId} = attributes;
	const setEntryId = (entryId) => setAttributes({entryId});
	const setFormId = (formId) => setAttributes({formId});
	let forms= {};
	let form = forms.hasOwnProperty(formId) ? forms[formId] : {};
	getForms(1)
		.then((response:Response) => {return response.json();})
		.then((response:KeyedFormCollection) => {
				forms = response;
				form = forms.hasOwnProperty(formId) ? forms[formId] : {};
		});


	if( isSelected ){
		return(
			<Fragment>
				<InspectorControls>
					<EntryControlsEdit
						instanceId={id}
						entryId={entryId}
						formId={formId}
						entries={getEntries(formId,1)}
						onSetForm={setFormId()}
						onSetEntry={setEntryId}
					/>
				</InspectorControls>
					<EntryInlineEdit
						instanceId={id}
						entryId={entryId}
						formId={formId}
						entries={getEntries(formId,1)}
						onSetForm={setFormId()}
						onSetEntry={setEntryId}
					/>
			</Fragment>
		);
	}

	return <EntryDisplay
		currentEntryId={entryId}
		entries={entries}
		getCurrentEntry={entries[entryId]}
		form={form}
	/>
};
