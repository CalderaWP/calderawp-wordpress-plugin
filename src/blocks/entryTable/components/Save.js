import {Title} from "./Title";
import React from "react";

export const ENTRY_TABLE_BLOCK_CLASSNAME = 'caldera-forms-entry-table-block';
export const Save =  ({attributes}) => {
	const {title,formId} = attributes;
	return <div class={ENTRY_TABLE_BLOCK_CLASSNAME} data-form={formId}>
		<Title title={title}/>
	</div>;
};

