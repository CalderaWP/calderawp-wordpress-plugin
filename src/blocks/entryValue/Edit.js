/** globals CF_ADMIN */
// @flow

import React, {Fragment, createElement} from 'react';
import Display from "./Display";
import {
	EntryControlsEdit,
	EntryInlineEdit
} from '../../components/Entry'
import {findFormById} from "../../components/Entry/findFormById";
import {InspectorControls} from '@wordpress/editor';
import Controls from "../../components/Entry/Edit/Controls";
import {getFieldFromCollection} from "../../components/EntryViewer/EntryHeaders";

import * as cfApi from "@caldera-labs/api-client";
import {formsAdminApiClient} from "../../../wp-content/plugins/caldera-forms/clients/state/api/apiClients";
const API_URL = CF_ADMIN.api.root.replace(/\/$/, "");
const NONCE = CF_ADMIN.api.nonce;

formsAdminApiClient.setNonce(NONCE);
export const entryApiClient = new cfApi.EntriesClient(API_URL);
entryApiClient.setNonce(NONCE);

function getForms(page: number): Promise<any> {
	return formsAdminApiClient.makeRequest('forms', {full: true, page});
}

function getEntries(formId: string, page: number): Promise<any> {
	return entryApiClient.makeRequest(`entries/${formId}`, {page});
}


let entry = {};
let form = {};
let entries = {};

export default function Edit(props) {
	const {attributes, setAttributes, isSelected, className, id} = props;
	const {fieldId, entryId, formId, before, after} = attributes;
	const setBefore = (before) => setAttributes({before});
	const setAfter = (after) => setAttributes({after});
	const setFieldId = (fieldId) => {
		let update : {
			before?:string,
			fieldId:string
		} = {fieldId};
		if( ! before ){
			const formFields = Object.assign( form.field_details.entry_list, form.field_details.order );
			const field = getFieldFromCollection(formFields,fieldId);
			if( field ){
				update.before= field.label + ' : ';
			}
		}
		setAttributes(update);
	}

	const setEntryId = (entryId) => {
		setAttributes({entryId});
	};
	const setFormId = (formId) => {
		getEntries(formId, 1)
			.then(response => response.json())
			.then(response => {
				entries = Object.values(response);
				if (entries.hasOwnProperty(entryId)) {
					entry = entries[entryId];
				}
				form = findFormById(forms, formId)
				setAttributes({formId});
			}).catch(e => {
			entries = {};
			entry = {};
			setAttributes({formId});

		});
	};


	let forms = CF_ADMIN ? CF_ADMIN.forms : {};
	forms = Object.values(forms);

	if ('object' !== typeof form) {
		form  = findFormById(forms, formId)
	}

	const editProps = {
		entryId: entryId,
		entries: entries,
		onSetEntry: setEntryId,
		formId: formId,
		forms: forms,
		form: form,
		onSetForm: setFormId,
		instanceId: id,
		onSetField: setFieldId,
		fieldId: fieldId,
		entryFieldId: fieldId,
		before: before,
		after: after,
		setBefore: setBefore,
		setAfter: setAfter,
	};


	const elements = [];
	if (isSelected) {
		elements.push(createElement(EntryInlineEdit, editProps));
	} else {
		if (entryId) {
			elements.push(
				<Display
					form={form}
					currentEntryId={entryId}
					entryId={entryId}
					entry={entry}
					formId={formId}
					entryFieldId={fieldId}
					before={before}
					after={after}
				/>
			)
		}
	}

	elements.push(
		createElement(
			InspectorControls,
			{}, [
				createElement(Controls, editProps)
			]
		)
	);


	return React.createElement(Fragment, {}, elements);


};
