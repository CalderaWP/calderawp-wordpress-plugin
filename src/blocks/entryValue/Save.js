
import {createElement} from '@wordpress/element';
import React from 'react';


export default function Save() {
	return null;

}