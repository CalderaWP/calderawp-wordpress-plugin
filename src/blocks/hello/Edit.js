import {
	HelloDisplay,
	HelloControlsEdit,
	HelloInlineEdit
} from '../../components/Hello';
import {Fragment,createElement} from '@wordpress/element'
import {InspectorControls} from '@wordpress/editor'
import type {GutenbergEditProps} from "../types";
import React from 'react';

export default function Edit(props: GutenbergEditProps) {
	const {attributes, setAttributes, isSelected, className} = props;
	const {name,salutation} = attributes;
	const onChangeSalutation = (salutation) => {setAttributes({salutation})};
	const onChangeName = (name) => {setAttributes({name})};
	if( isSelected ){
		return(
			<Fragment>
				<InspectorControls>
					<HelloControlsEdit
						name={name}
						salutation={salutation}
						onChangeSalutation={onChangeSalutation}
					/>
				</InspectorControls>
				<HelloInlineEdit
					name={name}
					salutation={salutation}
					onChangeName={onChangeName}
				/>
			</Fragment>
		);
	}

	return <HelloDisplay
		name={name}
		salutation={salutation}
	/>

}