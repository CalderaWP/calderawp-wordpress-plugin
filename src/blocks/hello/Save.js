import "babel-polyfill"
import {
	HelloDisplay,

} from '../../components/Hello';
import {Fragment,createElement} from '@wordpress/element';
import React from 'react';
import type {GutenbergSaveProps} from "../types";

export default function (props: GutenbergSaveProps) {
	const {attributes} = props;
	const {name,salutation} = attributes;

	return <HelloDisplay
		name={name}
		salutation={salutation}
	/>

}