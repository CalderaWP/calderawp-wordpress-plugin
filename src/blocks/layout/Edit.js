import {
	HelloDisplay,
	HelloControlsEdit,
	HelloInlineEdit
} from '../../components/Hello';
import {Fragment,createElement} from '@wordpress/element'
import {
	InspectorControls,
	//InnerBlocks
} from '@wordpress/editor';
const {InnerBlocks} = wp.editor;
import {registerBlockType} from '@wordpress/blocks';
import type {GutenbergEditProps} from "../types";
import React from 'react';


let fields = [
	{
		ID: 'fld10',
		label: 'Field One',
		type: 'text',
		value: 'value'
	},
	{
		ID: 'fld11',
		label: 'Field Two',
		type: 'file',
		value: 'picture'
	}
];

fields.forEach( field => {
	field.namespacedName =`something/${field.id}`
	registerBlockType( field.namespacedName, {
		title: field.label,
		// Only allow in a registered parent-block:
		parent: [ 'ibenic/parent-block' ],

	});
});




export default function Edit(props: GutenbergEditProps) {
	const {attributes, setAttributes, isSelected, className} = props;
	const {name,salutation} = attributes;
	const onChangeSalutation = (salutation) => {setAttributes({salutation})};
	const onChangeName = (name) => {setAttributes({name})};


	let allowedBlocks = [ 'core/image', 'core/paragraph' ];

	let template = [
		[ 'core/paragraph', { placeholder: 'Enter side content...' } ],
		[ 'core/paragraph', { placeholder: 'Second Content' } ]
	];
	fields.forEach( field => {
		allowedBlocks.push(field.namespacedName)
		template.push(field.namespacedName)
	});
	return (
		<div className={ className }>
			<InnerBlocks
				allowedBlocks={ allowedBlocks }
				template={template}
				templateLock={false}
			/>
		</div>
	);

}
