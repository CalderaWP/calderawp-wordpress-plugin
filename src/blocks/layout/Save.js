
import {Fragment,createElement} from '@wordpress/element';
import React from 'react';
import type {GutenbergSaveProps} from "../types";
const {InnerBlocks} = wp.editor;

export default function Save(props: GutenbergSaveProps) {


	return <InnerBlocks.Content />

}
