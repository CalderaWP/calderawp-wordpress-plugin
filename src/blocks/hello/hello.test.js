import '../../jest/matchMedia.mock'
import renderer from "react-test-renderer";
import React from "react";
import Save from './Save';
import Edit from './Edit';
import {InspectorControls} from '@wordpress/editor';
import Enzyme, {mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {componentClassName} from "../../control-factory/componentClassName";
Enzyme.configure({adapter: new Adapter()});

const attributeDefaults = {
	name: 'Roy',
	salutation: 'Hi'
};

describe('Edit callback for block', () => {


	let attributes = {};

	function setAttributes(newValues) {
		attributes = {
			...attributes,
			...newValues
		}
	}


	beforeEach(() => {
		attributes = attributeDefaults;
	});

	let className = 'a';

	it('mock set attributes works as expected', () => {
		setAttributes({name:'Mike'});
		expect( attributes.name).toBe( 'Mike' );
		expect( attributes.salutation).toBe( 'Hi' );
	});

	it('mock set attributes resets as expected', () => {
		setAttributes({salutation:'Hola'});
		expect( attributes.name).toBe( 'Roy' );
		expect( attributes.salutation).toBe( 'Hola' );
	});

	it( 'Matches snapshot when selected',() => {
		expect(
			renderer.create(
				<React.Fragment>
					<Edit
						attributes={attributeDefaults}
						setAttributes={setAttributes}
						className={'a'}
						isSelected={true}
					/>
					<InspectorControls.Slot />
				</React.Fragment>

			).toJSON()
		).toMatchSnapshot()
	});

	it( 'Matches snapshot when not selected',() => {
		expect(
			renderer.create(
				<React.Fragment>
					<Edit
						attributes={attributeDefaults}
						setAttributes={setAttributes}
						className={'a'}
						isSelected={false}
					/>
					<InspectorControls.Slot />
				</React.Fragment>

			).toJSON()
		).toMatchSnapshot()
	});


	it( 'Has usable name control',() => {
		const className = componentClassName('hello', 'name', 'edit');
		const component = mount(
			<React.Fragment>
				<Edit
					attributes={attributeDefaults}
					setAttributes={setAttributes}
					className={'a'}
					isSelected={true}
				/>
				<InspectorControls.Slot />
			</React.Fragment>
		);


		const event = {target: {value: "Mike"}};
		const input = component.children().find('.'+className).children().find('input');
		expect(input.length).toBe(1);
		input.simulate('change', event);
		expect(attributes.name).toBe('Mike');
	});






});

describe( 'Save callback for block', () => {
	it( 'Matches snapshot',() => {
		expect(
			renderer.create(
				<Save attributes={attributeDefaults} className={'a'}/>
			).toJSON()
		).toMatchSnapshot()
	})
});