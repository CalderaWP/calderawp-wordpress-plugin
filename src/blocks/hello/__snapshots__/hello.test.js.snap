// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit callback for block Matches snapshot when selected 1`] = `
<div
  className="components-base-control caldera-name-hello-edit"
>
  <div
    className="components-base-control__field"
  >
    <input
      aria-describedby={undefined}
      className="components-text-control__input"
      id="inspector-text-control-0"
      onChange={[Function]}
      type="text"
      value="Roy"
    />
  </div>
</div>
`;

exports[`Save callback for block Matches snapshot 1`] = `
<div
  className="caldera-salutation--name-hello-display"
>
  Hi
   
  Roy
</div>
`;
